src/Users/UserStatistics.php

```diff
@@ -0,0 +1,341 @@
+                'desc'  => $categoryDesc,
+            ],
+            'category_type' => (object) [
+                'value' => $typeCode,
+                'desc'  => $typeDesc,
+            ],
+            'statistic_note' => $note,
+            'segment_type'   => $segment_type,
+        ];
+
+        # Add the object to the user
+        $this->data->user_statistic[] = $stat_obj;
+        return;
+    }
+
+    /**
+    * Add a user statistic based upon existing Types and Categories.
+    *
+    * @param string $typeCode code of the category type.
+    * @param string $categoryCode code of the statistical category.
+    * @param string $segmentType either ("Internal" or "External")
+    * @param string $note
+    */
+    public function addStatistic($typeCode,$categoryCode,$segmentType = 'External',$note = '')
+    {
+        /* Logic: 
+           Lookup both $typeCode and $categoryCode in codetable.
+           Obtain already configured description from both code tables.
+           If found, add the statistic.
+        
+           Code Tables used:
+             Code Table; UserStatisticalTypes
+             Code Table: UserStatCategories
+        */
+
+        # Get the Type and Category Descriptions for the codes from the code tables.
+        $stat = $this->client->conf->codetables->get('UserStatisticalTypes')->getRowByCode($typeCode);
+        $typeDesc = $stat[0]->description;
+
+        $stat = $this->client->conf->codetables->get('UserStatCategories')->getRowByCode($categoryCode);
+        $categoryDesc = $stat[0]->description;
+        
+        # Add the statistic:
+        $this->addStatisticRaw($typeCode,$typeDesc,$categoryCode,$categoryDesc,$segmentType,$note);
+
+        return;
+    }
+
+    /**
+    * Delete a user statistic.
+    *
+    * @param string $typeCode code of the category type.
+    * @param string $categoryCode code of the statistical category.
+    */
+    public function removeStatistic($typeCode,$categoryCode)
+    {
+        foreach($this->data->user_statistic as $key => $row) {
+            if (($row->category_type->value == $typeCode) && ($row->statistic_category->value == $categoryCode)) {
+                array_splice($this->data->user_statistic, $key, 1);
+            }
+        }
+        return;
+    }
+
+    /**
+    * Delete all user statistics.
+    *
+    */
+    public function removeAllStatistics()
+    {
+        foreach($this->data->user_statistic as $key => $row) {
+            array_splice($this->data->user_statistic, $key, 1);
+        }
+        return;
+    }
+
+    /**
+    * Update a user statistic.
+    *
+    * @param string $fromTypeCode 
+    * @param string $fromCategoryCode
+    * @param string $toTypeCode
+    * @param string $toCategoryCode
+    * @param string $segmentType ("Internal" or "External")
+    * @param string $note
+    */
+    public function updateStatistic($fromTypeCode,$fromCategoryCode,$toTypeCode,$toCategoryCode,$segmentType = "External" ,$note = '')
```

Comment:
The update method is better as a member of `Statistic` rather than `Statistics`.